### PR TITLE
Add support for HTTP MERGE request generation

### PR DESCRIPTION
**Pull request by prashdsouza**
For Merge requests

corresponding changes have been in FioriDAST
I will email you the PR for FioriDAST

test sample for this implementation
image

Please also create a new version on merge to master.

### CODE REVIEW

File: odfuzz/entities.py [hunk 1]
```diff
@@ -144,7 +144,7 @@ def _append_queryable(self, query_group_data):
         # TODO REFACTOR DRY this method is direct copypaste from DispatchedBuilder just to have a prototype for integration. Intentionally no abstract class at the moment.
         if Config.fuzzer.http_method_enabled != "POST":
             self._append_corresponding_queryable(QueryGroupSingle(query_group_data))
-        if Config.fuzzer.http_method_enabled != "PUT":
+        if Config.fuzzer.http_method_enabled != "PUT" and Config.fuzzer.http_method_enabled != "MERGE":
```
Comment (phanak-sap) [inline odfuzz/entities.py:147]: I think this should be OR logic, not AND. 

You can have only one method generation enabled in the config and it is expected that will work, unrelated to the other method.

File: odfuzz/fuzzer.py [hunk 2]
```diff
@@ -368,6 +368,22 @@ def generate_body(self,accessible_entity,key_pairs):
                 except:
                     pass
                 body[prprty._name] = generated_body
+        elif Config.fuzzer.http_method_enabled == "MERGE":
+            properties = deepcopy(accessible_entity.entity_set.entity_type._properties)
```
Comment (phanak-sap) [inline odfuzz/fuzzer.py:372]: What is the purpose of the deepcopy..? Why the PUT and POST are same and just generating stuff, while MERGE needs to copy the structure and than call some randomized stuff on it? Why the body content is dependent on random? What if we want to generate MERGE requests with specific properties, not select them on random?

Each if/elif block seems to me needs some kind of comments explaining the rationale behind. Consider the blocks to be separate functions (which they will probably will be refactored to in the future, with its unit tests) and think of pydocs for them. Functions would probably serve better if we would like to - and we will - to get rid of random.something() scattered trough the codebase and have the Requests generation better configurable, say to some TestStrategy.

File: odfuzz/entities.py (hunk 1 above)
Comment (phanak-sap) [reply odfuzz/entities.py:147]: I totally misunderstood the logic and purpose due inverted != logic. 
Agreed in person to rewrite this part to normal IF a == b logic, so the method is more readable and maintainable when we add new HTTP method support

File: odfuzz/fuzzer.py (hunk 2 above)
Comment (AlapanGhosh) [reply odfuzz/fuzzer.py]: addressed as discussed

File: odfuzz/entities.py (hunk 1 above)
Comment (AlapanGhosh) [reply odfuzz/entities.py]: addressed

File: tests/integration/url_generator_only/test_url_generator_only.py [hunk 3]
```diff
@@ -90,3 +91,79 @@ def test_direct_builder_http_delete():
...
+def test_direct_builder_http_merge_body():
+    random.seed(20)
+    path_to_metadata = Path(__file__).parent.joinpath("metadata-northwind-v2.xml")
+    metadata_file_contents = path_to_metadata.read_bytes()
+    restrictions = RestrictionsGroup(None)
+    dir_builder = DirectBuilder(metadata_file_contents, restrictions,"MERGE")
+    dir_entities = dir_builder.build()
+    queryable_factory = SingleQueryable
+    body_list = []
+    body_list.clear()
```
Comment (phanak-sap) [inline tests/integration/url_generator_only/test_url_generator_only.py:162]: The initialization part seems to be same for each test and copypasted. Please extract to function so the tests themselves are shorter.

**Comment by phanak-sap** [comment]
Small notes, generally OK. Don't forget to update changelog.md with the new feature as well pls.

**Pull request merged**